### ISSUE TITLE
contract/collection level metadata

### ISSUE THREAD

**Issue by btspoony** [onflow/flow-nft#62]
## Issue To Be Solved

At present, when Flow wallets or DApps integrate an NFT Collection, what the basic information of NFT collections (like display name, description, etc.) needs to be collected case by case.

Have we considered proposing a `FLIP` that implements a standard  for the definition of contract or collection level metadata?  
Reference: https://docs.opensea.io/docs/contract-level-metadata

If such an standard exists, the work will be handed over to the NFT contract developers who will implement Views like `GeneralDisplay`, `BloctoDisplay`, `AlchemyDisplay`, `NewWalletDisplay` with their collection level information(contract name, display name, description, banner image, etc.).

**Comment by bjartek** [comment]
This is what alchemy needs for their impl so that should be considered. https://github.com/alchemyplatform/alchemy-flow-contracts/blob/main/src/cadence/mainnet/getNFTs.cdc#L90

**Comment by lmcmz** [comment]
Yeah, this a good idea. I also made a similar suggestion in their repo, but haven't got any reply on this yet.  :/
https://github.com/alchemyplatform/alchemy-flow-contracts/issues/35

**Comment by joshuahannan** [comment]
Can you elaborate on what kind of info would be included in this view? Seems to me like most of this is already included in other views

**Comment by btspoony** [comment]
> Can you elaborate on what kind of info would be included in this view? Seems to me like most of this is already included in other views

This does not refer to NFT information, but to **NFT collection** information. 
For example:
- **name** of the collection
- **description** of the collection
- **logo image** of the collection
- **official website** of the collection
- **storage_path** of the collection
- **public_path** of the collection
- **public_collection_name** of the collection

which can be used in https://github.com/alchemyplatform/alchemy-flow-contracts/blob/e44cea7ebe90d1b289f999c2b40c4ed4393ffd0a/src/cadence/mainnet/getNFTs.cdc#L117
and also can be used for DApps displaying for the NFT collection (not each NFT)

**Comment by joshuahannan** [comment]
If you have a reference to a collection, isn't a lot of this information already available? You had to use the name and paths to get it already. 

Also, how would you propose we integrate this with the standard? This doesn't fit with the metadata standard because that is for individual NFTs. Would you propose adding another contract interface to implement?

**Comment by btspoony** [comment]
> * 
> * **storage_path** of the collection
> * **public_path** of the collection
> * **public_collection_name** of the collection

Yeah, I mean it should be another contract interface standard, not to fit with the current individual NFT metadata standard.

If DApps want to list all NFT collections in your account, it will be easy to check if the nft collection exists in your account and display basic collection information.

**Comment by btspoony** [comment]
> If you have a reference to a collection, isn't a lot of this information already available? You had to use the name and paths to get it already.
> 
> Also, how would you propose we integrate this with the standard? This doesn't fit with the metadata standard because that is for individual NFTs. Would you propose adding another contract interface to implement?

Yeah, I mean it should be another contract interface standard, not to fit with the current individual NFT metadata standard.

If DApps want to list all NFT collections in your account, it will be easy to check if the nft collection exists in your account and display basic collection information.

**Comment by bluesign** [comment]
This can be handled by current metadataviews as they work on any resource. 

Just making a CollectionView should be enough.

**Comment by albeethekid** [comment]
For the images/media it would be great to be able to support several images (banner, square icon etc). I believe @psiemens had some ideas here. 

Having a dynamically driven CollectionView would be great for those contracts (Rarible for example) that have been used for multiple NFT collections.

**Comment by aishairzay** [comment]
@btspoony 
>Or we can add a method to get NFTCollectionData at the Contract level?

The reason it would be tough to place the NFT collection-level data on the contract level is because many NFTs on Flow today use a single contract to distribute many different NFT collections. This includes collections such as Rarible and Starly. 
With the solution in the following PRs, we can support those types of contracts: https://github.com/onflow/flow-nft/pull/82 and https://github.com/onflow/flow-nft/pull/83

> Or we will be using the generic Collection with same PublicPath in the future, according to [Streamline-token-standards](https://forum.onflow.org/t/streamlined-token-standards-proposal/3075), and still hard coding for the paths?

Yeah, that's what i'm thinking - the problem of discovering NFT collections is not solved by these new views, but this does  allow for other solutions to come up. Which can be in the form of the new streamlined token standard you linked or also cadence storage discovery APIs which IIRC are not too far out from being a reality. Our team is also working on an on-chain, catalog of NFT collections as a temporary solution until one of the aforementioned are available to have an easy way to retrieve all paths of relevant NFT collections.

I believe the NFTCollectionData in https://github.com/onflow/flow-nft/pull/82 might be able to be deprecated someday if we go forward with the streamline-token-standards new NFT standard as well.
